### PR TITLE
Update sources

### PR DESCRIPTION
**Pull request by anmonteiro**
:robot_face: Updating sources to the latest version.

#### Commits touching OCaml packages:
* <a href="https://github.com/NixOS/nixpkgs/commit/6043e3af4ec34a42d1f45620527eb3081e5b564a"><pre>ocamlPackages.findlib: detect conflicts of version early

just like pythonCatchConflictHook</pre></a>
* <a href="https://github.com/NixOS/nixpkgs/commit/8281faf75a1ef02d021f7f8ca9cc50c9997a0616"><pre>ocamlPackages.javalib: 3.2.1 → 3.2.2</pre></a>
* <a href="https://github.com/NixOS/nixpkgs/commit/3968d0e5b98331c60c8fdebfb3798496aa39e193"><pre>ocamlPackages.sawja: 1.5.11 → 1.5.12</pre></a>
* <a href="https://github.com/NixOS/nixpkgs/commit/b55a8d3ad6bfd031230e654e63ff406facd2976d"><pre>Merge pull request #205646 from symphorien/ocamlconflict

ocamlPackages.findlib: detect conflicts of version early</pre></a>
* <a href="https://github.com/NixOS/nixpkgs/commit/9469779ebefbaeb96a3b0eaa2f87c9195b089bee"><pre>Merge pull request #212910 from vbgl/ocaml-javalib-3.2.2

ocamlPackages.javalib: 3.2.1 → 3.2.2; sawja: 1.5.11 → 1.5.12</pre></a>
* <a href="https://github.com/NixOS/nixpkgs/commit/7a7c9f938cdae63dad5ca033966ad8fe3d25f050"><pre>ocamlPackages.mdx: 2.1.0 → 2.2.1</pre></a>
* <a href="https://github.com/NixOS/nixpkgs/commit/5d5aa12849b021081ea8a2eb704e5db1f462e3ae"><pre>Merge pull request #213102 from vbgl/ocaml-mdx-2.2.1

ocamlPackages.mdx: 2.1.0 → 2.2.1</pre></a>
* <a href="https://github.com/NixOS/nixpkgs/commit/4c4b377a23282b516bc991f0bc928d91dc28994a"><pre>Merge pull request #212407 from NobbZ/hledger-utils

hledger-utils: init at 1.12.1 (also init so far unpackaged dependencies)</pre></a>
* <a href="https://github.com/NixOS/nixpkgs/commit/4c4b377a23282b516bc991f0bc928d91dc28994a"><pre>Merge pull request #212407 from NobbZ/hledger-utils

hledger-utils: init at 1.12.1 (also init so far unpackaged dependencies)</pre></a>
* <a href="https://github.com/NixOS/nixpkgs/commit/4c4b377a23282b516bc991f0bc928d91dc28994a"><pre>Merge pull request #212407 from NobbZ/hledger-utils

hledger-utils: init at 1.12.1 (also init so far unpackaged dependencies)</pre></a>
* <a href="https://github.com/NixOS/nixpkgs/commit/4c4b377a23282b516bc991f0bc928d91dc28994a"><pre>Merge pull request #212407 from NobbZ/hledger-utils

hledger-utils: init at 1.12.1 (also init so far unpackaged dependencies)</pre></a>
* <a href="https://github.com/NixOS/nixpkgs/commit/4c4b377a23282b516bc991f0bc928d91dc28994a"><pre>Merge pull request #212407 from NobbZ/hledger-utils

hledger-utils: init at 1.12.1 (also init so far unpackaged dependencies)</pre></a>

#### Diff URL: https://github.com/NixOS/nixpkgs/compare/3875a3e3e1ca2fa7e17fc45a1dbf92551f1b9db7...4c4b377a23282b516bc991f0bc928d91dc28994a

### CODE REVIEW

**Comment by ulrikstrid** [comment]
I will try to upstream the last patch, I think those libraries are the ones that are OK to be in both right?

File: cross/ocaml.nix [hunk 1]
```diff
@@ -151,6 +151,44 @@ in
...
+              export OCAMLFIND_DESTDIR="''$out/lib/ocaml/${oself.ocaml.version}/site-lib/"
+          }
+          createOcamlDestDir () {
+              if test -n "''${createFindlibDestdir-}"; then
+                mkdir -p $OCAMLFIND_DESTDIR
+              fi
+          }
+          detectOcamlConflicts () {
+            local conflict
+            conflict="$(ocamlfind list |& grep "has multiple definitions" | grep -vE "bigarray|unix|str|stdlib|compiler-libs|threads|bytes|dynlink|findlib" || true)"
```
[the remark this answers is not in the document]
Comment (anmonteiro) [reply cross/ocaml.nix:174]: `||` true seems to always make it pass, no?

[the remark this answers is not in the document]
Comment (ulrikstrid) [reply cross/ocaml.nix:174]: This is taken from upstream so I assume it's correct :joy:

**Comment by anmonteiro** [comment]
OK let's land this until upstream makes up their mind.

**Pull request merged**